pkg/delete/internal/classic/delete.go

```diff
@@ -21,106 +21,140 @@ import (
-		scopedMapped := map[string][]pointer.DeletePointer{}
-		for _, entry := range entries {
-			scopedMapped[entry.Scope] = append(scopedMapped[entry.Scope], entry)
+		if err := client.DeleteConfigById(*a, resolvedID); err != nil && !is404(err) {
+			log.WithFields(field.Coordinate(e.coordinate)).Error("failed to delete config: %w")
+			err2 = errors.Join(err2, err)
 		}
+	}
 
-		for scope, scopeEntries := range scopedMapped {
-			a := theApi.Resolve(scope)
+	return err2
+}
 
-			var values []dtclient.Value
-			values, err = client.ListConfigs(ctx, a)
+func convertToEntry(p pointer.DeletePointer, theAPI api.API) *entry {
+	var e *entry
+	e = &entry{
+		identifier: p.Identifier,
+		coordinate: p.AsCoordinate(),
+	}
+	if theAPI.Parent != nil {
+		e.parent = &entry{
+			identifier: p.Scope,
+			coordinate: p.AsCoordinate(),
+		}
+	}
+	return e
+}
 
-			if err != nil {
-				var respErr rest.RespError
-				if errors.As(err, &respErr) && respErr.StatusCode == http.StatusNotFound {
-					logger.Debug("No config of type %q found to delete for scope %q :%s", theApi.ID, scope, respErr.Body)
-					err = nil
-				} else {
-					logger.WithFields(field.Error(err)).Error("Failed to fetch existing configs for api %q (scope: %s): %w", a.ID, scope, err)
-					deleteErrs++
-				}
-				continue
-			}
+func is404(err error) bool {
+	var v rest.RespError
+	if errors.As(err, &v) {
+		return v.StatusCode == http.StatusNotFound
+	}
+	return false
+}
 
-			var vals []deleteValue
-			vals, err = filterValuesToDelete(logger, scopeEntries, values, theApi.ID)
+// resolveIdentifier get the actual ID from DT and update entries with it
+func resolveIdentifier(ctx context.Context, client dtclient.Client, theAPI *api.API, e *entry) (*api.API, string, error) {
+	var err error
 
-			delValues = append(delValues, vals...)
+	if theAPI.Parent != nil {
+		if _, _, err := resolveIdentifier(ctx, client, theAPI.Parent, e.parent); err != nil && !is404(err) {
+			return nil, "", err
 		}
 	}
 
-	if err != nil {
-		deleteErrs++
+	if theAPI.Parent != nil && e.parentID() == "" {
+		log.WithFields(field.Coordinate(e.coordinate)).Debug("parent config gone")
+		return &api.API{}, "", nil
 	}
 
-	if len(delValues) == 0 {
-		logger.Debug("No values found to delete for type %q.", targetApi)
-		return err
+	a := theAPI.Resolve(e.parentID())
+	knownValues, err := client.ListConfigs(ctx, a)
+	if err != nil {
+		return nil, "", err
 	}
 
-	logger.Info("Deleting %d config(s) of type %q...", len(delValues), theApi.ID)
+	id, err := findUniqueID(knownValues, e.identifier)
+	if err != nil {
+		return nil, "", err
+	}
 
-	for _, v := range delValues {
-		vLog := logger.WithFields(field.Coordinate(v.AsCoordinate()), field.F("value", v))
+	e.resolvedID = id
+	return &a, id, nil
+}
 
-		a := theApi
-		if a.HasParent() {
-			a = a.Resolve(v.DeletePointer.Scope)
-		}
+func findUniqueID(knownValues []dtclient.Value, identifier string) (string, error) {
```

Comment:
How does deletion work? For a given API is it always by name or ID? Or is it really mixed?